stdlib/format.mli

```diff
@@ -107,6 +103,446 @@
+
+within a "v" box:
+
+{[
+--
+--
+--
+]}
+
+within a "hv" box:
+{[
+-- -- --
+]}
+
+or, according to the remaining room on the line:
+
+{[
+--
+--
+--
+]}
+
+and similarly for "hov" boxes.
+
+Generally speaking, a printing routine using "format", should not directly
+output white spaces: the routine should use break hints instead. (For instance
+[print_space ()] that is a convenient abbreviation for [print_break 1 0] and
+outputs a single space or break the line.)
+
+
+
+{2 Indentation of new lines}
+
+The user gets 2 ways to fix the indentation of new lines:
+
+{b When defining the box}: when you open a box, you can fix the indentation
+added to each new line opened within that box.
+
+For instance: [open_hovbox 1] opens a "hov" box with new lines indented 1 more
+than the initial indentation of the box. With output "---\[--b--b--b--", we
+get:
+
+{[
+---\[--b--b
+     --b--
+]}
+
+with open_hovbox 2, we get
+
+{[
+---\[--b--b
+      --b--
+]}
+
+Note: the \[ sign in the display is not visible on the screen, it is just there
+to materialise the aperture of the pretty-printing box. Last "screen" stands
+for:
+
+{[
+-----b--b
+     --b--
+]}
+
+{b When defining the break that makes the new line}. As said above, you output
+a break hint using [print_break sp indent]. The [indent] integer is used to fix
+the additional indentation of the new line. Namely, it is added to the default
+indentation offset of the box where the break occurs.
+
+For instance, if \[ stands for the opening of a "hov" box with 1 as extra
+indentation (as obtained by [open_hovbox 1]), and b is [print_break 1 2], then
+  from output "---\[--b--b--b--", we get:
+
+{[
+   ---\[-- --
+         --
+         --
+]}
+
+{2 Refinement on "hov" boxes}
+
+
+The "hov" box type is refined into two categories.
+
+- {b the vertical or horizontal {i packing} box} (as obtained by the
+{!open_hovbox} procedure): break hints are used to cut the line when there is no
+more room on the line; no new line occurs if there is enough room on the line.
+- {b vertical or horizontal {i structural} box} (as obtained by the {!open_box}
+procedure): similar to the "hov" packing box, the break hints are used to cut
+the line when there is no more room on the line; in addition, break hints that
+can show the box structure lead to new lines even if there is enough room on
+the current line.
+
+
+The difference between a packing and a structural "hov" box is shown by a
+routine that closes boxes and parentheses at the end of printing: with packing
+boxes, the closure of boxes and parentheses do not lead to new lines if there
+is enough room on the line, whereas with structural boxes each break hint will
+lead to a new line. For instance, when printing
+"\[(---\[(----\[(---b)\]b)\]b)\]", where "b" is a break hint without extra
+indentation ([print_cut ()]). If "\[" means opening of a packing "hov" box
+({!open_hovbox}), "\[(---\[(----\[(---b)\]b)\]b)\]" is printed as follows:
+
+{[
+(---
+ (----
+  (---)))
+]}
```

Comment:
This example and the one below are confusing since there is no break hints before the inner `(`s, the hov boxes are supposed to be open without any indentations. In fact, the only way to get the expected output is to add some indentation and twiddle with the maximum indentation limit (which is not explained at all in this tutorial!):
```ocaml
Format.set_geometry ~max_indent:3 ~margin:10;
Format.printf 
"@[<hov 1>(---@[<hov 1>(----@[<hov 1>(---@,)@]@,)@,@]@,)@]@.";;
```

Reply:
Note that this remark can be fixed by updating the introduction to mention the box indentation and adding break hints before opening the boxes rather than relying on the maximum indentation:

> For instance, when printing
"\[(---b\[(----b\[(---b)\]b)\]b)\]", where "b" is a break hint without extra
indentation ([print_cut ()]). If "\[" means opening of a packing "hov" box
with an indentation of 1,
({!open_hovbox}), "\[(---b\[(----b\[(---b)\]b)\]b)\]" is printed as follows:

or using format string

> For instance, when printing
"@[<hov 1>(---@,@[<hov 1>(----@,@[<hov 1>(---@,)@]@,)@]@,)@]"